multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2057,6 +2066,43 @@ def predict(
         if (as_pandas is None and isinstance(data, pd.DataFrame)) or as_pandas is True:
             pred = self._as_pandas(data=data, to_be_converted=pred)
 
+        if save_results:
+            if not result_path:
+                utcnow = datetime.utcnow()
+                timestamp = utcnow.strftime("%Y%m%d_%H%M%S")
+
+                result_path = f"Result{os.path.sep}ag-{timestamp}{os.path.sep}" \
+                              f"{self._hyperparameters['model.mmdet_image.checkpoint_name']}{os.path.sep}"
+                for i in range(1, 1000):
+                    try:
+                        os.makedirs(result_path, exist_ok=False)
+                        break
+                    except FileExistsError:
+                        logger.log(25, f"result_path {result_path} already exists")
+                else:
+                    raise RuntimeError("more than 1000 jobs launched in the same second")
+                logger.log(25, f'No result_path specified. Results will be saved in: "{result_path}"')
+                result_path = os.path.join(result_path, "results.ext")
```

Comment:
this may need to be refactored to init_result_path (or other names) and put in utils